stdlib/public/Darwin/Accelerate/Quadrature.swift

```diff
@@ -0,0 +1,214 @@
+        
+        self.integrand = integrand
+        
+        let byteCount = MemoryLayout<Quadrature>.size
+        let integrandPointer = UnsafeMutableRawPointer.allocate(byteCount: byteCount,
+                                                                alignment: 1)
+        integrandPointer.storeBytes(of: self,
+                                    as: Quadrature.self)
+        
+        var callback = quadrature_integrate_function(
+            fun: { (arg: UnsafeMutableRawPointer?,
+                n: Int,
+                x: UnsafePointer<Double>,
+                y: UnsafeMutablePointer<Double>
+                ) in
+                
+                guard let quadrature = arg?.load(as: Quadrature.self) else {
+                    return
+                }
+                
+                (0 ..< n).forEach { i in
+                    y[i] = quadrature.integrand(x[i])
+                }
+        },
+            fun_arg: integrandPointer)
+        
+        var status = QUADRATURE_SUCCESS
+        var estimatedAbsoluteError: Double = 0
+        var result: Double = 0
+        
+        withUnsafePointer(to: integrateOptions) { options in
+            result = quadrature_integrate(&callback,
+                                          interval.lowerBound,
+                                          interval.upperBound,
+                                          options,
+                                          &status,
+                                          &estimatedAbsoluteError,
+                                          0,
+                                          nil)
+        }
+        
+        if status == QUADRATURE_SUCCESS {
+            return .success(integralResult: result,
+                            estimatedAbsoluteError: estimatedAbsoluteError)
+        } else {
+            return .failure(error: Error(quadratureStatus: status))
+        }
+    }
+    
+    public enum Integrator {
+        /// Simple non-adaptive automatic integrator using Gauss-Kronrod-Patterson quadrature coefficients.
+        /// Evaluates 21, or 43, or 87 points in the interval until the requested accuracy is reached.
+        case qng
+        
+        /// Simple globally adaptive integrator.
+        /// Allows selection of the number of Gauss-Kronrod points used in each subinterval, and the max number of subintervals.
+        case qag(pointsPerInterval: QAGPointsPerInterval, maxIntervals: Int)
+        
+        /// Global adaptive quadrature based on 21-point or 15-point (if at least one bound is infinite) Gauss–Kronrod quadrature within each subinterval, with acceleration by Peter Wynn's epsilon algorithm.
+        /// If at least one of the interval bounds is infinite, this is equivalent to the QUADPACK QAGI routine. Otherwise, this is equivalent to the QUADPACK QAGS routine.
+        case qags(maxIntervals: Int)
+    }
+    
+    public enum QAGPointsPerInterval: Int {
+        case `default` = 0
```

Comment:
FYI: Raw-values do not affect layout for Swift enums. If you want something maximally-efficient (wrapping an integer) use a struct:

```
public struct QAGPointsPerInterval {
  public let points: Int
  private init(points: Int) { self.points = points }

  static let fifteen = QAGPointsPerInterval(points: 15)
  static let twentyOne = QAGPointsPerInterval(points: 21)
  // ..etc
}
```

Reply:
Implemented as a struct (pending push).